rmw_fastrtps_cpp/src/functions.cpp

```diff
@@ -919,6 +917,79 @@ rmw_node_t * rmw_create_node(const char * name, const char * namespace_, size_t
   return NULL;
 }
 
+bool rmw_get_security_file_paths(
+  std::array<std::string, 3> & security_files_paths, const char * node_secure_root)
+{
+  // here assume only 3 files for security
+  const char * file_names[3] = {"ca.cert.pem", "cert.pem", "key.pem"};
+  size_t num_files = 3;
+
+  const char * file_prefix = "file://";
+
+  std::string tmpstr;
+  for (size_t i = 0; i < num_files; i++) {
+    tmpstr = std::string(rcutils_join_path(node_secure_root, file_names[i]));
+    if (!rcutils_is_readable(tmpstr.c_str())) {
+      return false;
+    }
+    security_files_paths[i] = std::string(file_prefix + tmpstr);
+  }
+  return true;
+}
+
+rmw_node_t *
+rmw_create_node(
+  const char * name,
+  const char * namespace_,
+  size_t domain_id,
+  const rmw_node_security_options_t * security_options)
+{
+  if (!name) {
+    RMW_SET_ERROR_MSG("name is null");
+    return NULL;
+  }
+
+  ParticipantAttributes participantAttrs;
+  participantAttrs.rtps.builtin.domainId = static_cast<uint32_t>(domain_id);
+  participantAttrs.rtps.setName(name);
+
+  if (security_options->security_root_path) {
```

Comment:
You must check that `security_options` is not null before accessing its members.